nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/AbstractAzureDataLakeStorageProcessor.java

```diff
@@ -0,0 +1,154 @@
+        .required(true)
+        .build();
+
+    public static final PropertyDescriptor DIRECTORY = new PropertyDescriptor.Builder().name("Directory-name")
+        .displayName("Directory Name")
+        .description("Name of the Azure Storage Directory, It is assumed to be already existing")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(true)
+        .build();
+
+    public static final PropertyDescriptor PROP_SAS_TOKEN = new PropertyDescriptor.Builder()
+        .name("storage-sas-token").displayName("SAS Token")
+        .description("Shared Access Signature token, including the leading '?'. Specify either SAS Token (recommended) or Account Key. " +
+            "There are certain risks in allowing the SAS token to be stored as a flowfile " +
+            "attribute. While it does provide for a more flexible flow by allowing the account name to " +
+            "be fetched dynamically from a flowfile attribute, care must be taken to restrict access to " +
+            "the event provenance data (e.g. by strictly controlling the policies governing provenance for this Processor). " +
+            "In addition, the provenance repositories may be put on encrypted disk partitions.")
+        .required(false)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .sensitive(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor FILE = new PropertyDescriptor.Builder().name("file-name").displayName("File Name")
+        .description("The filename")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(true)
+        .defaultValue("nifi.${uuid}")
+        .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success").description(
+            "Files that have been successfully written to Azure storage are transferred to this relationship")
+        .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure").description(
+            "Files that could not be written to Azure storage for some reason are transferred to this relationship")
+        .build();
+
+    private static final List <PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(
+        Arrays.asList(AbstractAzureDataLakeStorageProcessor.ACCOUNT_KEY, AbstractAzureDataLakeStorageProcessor.ACCOUNT_NAME,
+            AbstractAzureDataLakeStorageProcessor.FILESYSTEM, AbstractAzureDataLakeStorageProcessor.DIRECTORY,
+            AbstractAzureDataLakeStorageProcessor.PROP_SAS_TOKEN, AbstractAzureDataLakeStorageProcessor.FILE));
+
+    private static final Set <Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet <>(
+        asList(

```

Comment:
Could you please use either `Arrays.asList` or `asList` (with static import) both in lines 113 and 118?